frontend/public/components/build-pipeline.scss

```diff
@@ -0,0 +1,334 @@
+$pipeline-aborted-color: $color-pf-black-300;
+$pipeline-border-color: $color-pf-black-300;
+$pipeline-failed-color: $color-pf-red-100;
+$pipeline-font-base: 12px;
+$pipeline-in-progress-color: $color-pf-blue-400;
+$pipeline-new-color: $color-pf-blue-100;
+$pipeline-pending-color: $color-pf-black-300;
+$pipeline-success-color: $color-pf-green-400;
+$pipeline-require-attention-color: $color-pf-gold-400;
+$pipeline-circle-animation-time: 0.35s;
+$pipeline-circle-diameter: 18px;
+$pipeline-line-border-width: 8px;
+$pipeline-circle-border-width: ($pipeline-line-border-width / 2);
+$pipeline-circle-radius: ($pipeline-circle-diameter / 2);
+$pipeline-inner-circle-animation-time: 0.1s;
+$pipeline-icon-animation-time: $pipeline-inner-circle-animation-time;
+$pipeline-inner-circle-color: #fff;
+$pipeline-line-animation-time: $pipeline-circle-animation-time;
+$pipeline-line-border-width: 8px;
+$pipeline-line-grow-animation-time: 0.5s;
+$pipeline-line-height: ($pipeline-line-border-width / 2);
+$pipeline-padding: 10px;
+$pipeline-progress-line: 100%;
+$pipeline-progress-rail-animation-time: 5s;
+$pipeline-semi-circle-animation-time: ($pipeline-circle-animation-time / 2);
+
+// Animations
+@keyframes build-progress-line {
+  to {width: $pipeline-progress-line}
+}
+
+@keyframes build-progress-rail {
+  to {transform: translateX(400%)}
+}
+
+@keyframes build-progress {
+  to {transform: rotate(180deg)}
+}
+
+@keyframes pipeline-stage-fadeIcon {
+  to {opacity: 1}
+}
+
+@keyframes pipeline-stage-fadeOut {
+  to {background-color: transparent}
+}
+
+.build-pipeline {
+  border: 1px solid $pipeline-border-color;
+	font-size: $pipeline-font-base;
+  margin-bottom: 30px;
+  &:first-child {
+    border-top-width: 1px;
+  }
+  // Switch summary from top to left placement
+  @media (min-width: 600px) {
+    display: flex;
+    flex: 1 1 0%;
+    flex-direction: row;
+  }
+}
+
+.build-pipeline__container {
+  flex: 1 1 auto;
+  overflow: hidden;
+}
+.build-pipeline__stages {
+  display: flex;
+  flex-wrap: wrap;
+  height: 100%;
+  padding: 0 ($pipeline-padding / 2);
+}
+
+.build-pipeline__stage {
+  // need to use flex-basis: auto and width because of IE11 bug
+  flex: 0 0 auto;
+  min-height: 96px;
+  padding: ($pipeline-padding + 5) $pipeline-padding  ($pipeline-padding * 3) $pipeline-padding;
+  position: relative;
+  width: 100%;
+  // add arrow after each stage
+  &:before {
+    bottom: 0;
+    color:  darken($pipeline-border-color, 5%);
+    content: '\2193';
+    font-size: 22px;
+    left: 0;
+    line-height: 1;
+    position: absolute;
+    right: 0;
+    text-align: center;
+  }
+  // hide arrow on last stage
+  &:last-child:before {
+    display: none;
+  }
+  @media (min-width: 480px) {
+    padding-right: ($pipeline-padding * 5.2);
+    padding-bottom: ($pipeline-padding + 5);
+    width: (100% / 3);
+    &:before {
+      bottom: auto;
+      content: '\2192';
+      left: auto;
+      right: 10px;
+      top: 37%;
+    }
+  }
+}
+
```

Comment:
Untested, but since we have that open topic regarding `sass` loops, An alternative could be to use a `sass` map, something like this:

```scss
// can also use a map, with key-value pairs
$pipeline-stages-per-row: (
  small: (
    column-no: 4,
    media-query-px: $screen-md-min
  )
  medium: (
    column-no: 4,
    media-query-px: $screen-lg-min
  )
  large: (
    column-no: 4,
    media-query-px: $screen-lg-min + 200
  )
  x-large: (
    column-no: 4,
    media-query-px: $screen-lg-min + 400
  )
)

// and use the map-get() function to get value by key...
@each $size, $map in $pipeline-stages-per-row {
  @media (min-width: map-get($map, media-query-px)) {
    .build-pipeline__stage {
      width: (100% / map-get($map, column-no));
    }
  }
}
```
`map-get($map, the-key-to-get)` is clearer than `nth($screenSize, someIndex)`, but the code is a bit longer.